From 5f925fe6231d0093f4292755cc1cf2b6374a1b9e Mon Sep 17 00:00:00 2001
From: Tomasz Figa <tfiga@chromium.org>
Date: Fri, 3 Jun 2016 16:09:00 +0900
Subject: [PATCH 515/517] CHROMIUM: gallium/dri: Advertise R8G8B8A8 and
 R8G8B8X8 DRI configs

Android requires these drawables and if hardware supports them and they
are already handled by other code, there is no reason not to advertise
them.

BUG=b:29036398
TEST=No significant regressions in Android dEQP

Change-Id: I14526b975bb4d0aa979bada13ab5360dc62ee887
Signed-off-by: Tomasz Figa <tfiga@chromium.org>
---
 src/gallium/state_trackers/dri/dri_screen.c | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/src/gallium/state_trackers/dri/dri_screen.c b/src/gallium/state_trackers/dri/dri_screen.c
index 2ac55c8..46cb70b 100644
--- a/src/gallium/state_trackers/dri/dri_screen.c
+++ b/src/gallium/state_trackers/dri/dri_screen.c
@@ -104,17 +104,33 @@ static const __DRIconfig **
 dri_fill_in_modes(struct dri_screen *screen)
 {
    static const mesa_format mesa_formats[] = {
+      MESA_FORMAT_R8G8B8A8_UNORM,
+      MESA_FORMAT_R8G8B8X8_UNORM,
       MESA_FORMAT_B8G8R8A8_UNORM,
       MESA_FORMAT_B8G8R8X8_UNORM,
+      MESA_FORMAT_R8G8B8A8_SRGB,
+      MESA_FORMAT_R8G8B8X8_SRGB,
       MESA_FORMAT_B8G8R8A8_SRGB,
       MESA_FORMAT_B8G8R8X8_SRGB,
+      MESA_FORMAT_R8G8B8A8_UNORM,
+      MESA_FORMAT_R8G8B8X8_UNORM,
+      MESA_FORMAT_R8G8B8A8_SRGB,
+      MESA_FORMAT_R8G8B8X8_SRGB,
       MESA_FORMAT_B5G6R5_UNORM,
    };
    static const enum pipe_format pipe_formats[] = {
+      PIPE_FORMAT_RGBA8888_UNORM,
+      PIPE_FORMAT_RGBX8888_UNORM,
       PIPE_FORMAT_BGRA8888_UNORM,
       PIPE_FORMAT_BGRX8888_UNORM,
+      PIPE_FORMAT_RGBA8888_SRGB,
+      PIPE_FORMAT_RGBX8888_SRGB,
       PIPE_FORMAT_BGRA8888_SRGB,
       PIPE_FORMAT_BGRX8888_SRGB,
+      PIPE_FORMAT_RGBA8888_UNORM,
+      PIPE_FORMAT_RGBX8888_UNORM,
+      PIPE_FORMAT_RGBA8888_SRGB,
+      PIPE_FORMAT_RGBX8888_SRGB,
       PIPE_FORMAT_B5G6R5_UNORM,
    };
    mesa_format format;
-- 
2.8.0.rc3.226.g39d4020

